docs/recipes/add-custom-chain.mdx

```diff
@@ -0,0 +1,163 @@
+---
+sidebar_position: 5
+title: Add a custom chain
+description: Learn how to add custom chains to your project.
+---
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+# Add a custom chain
+
+This recipe demonstrates how to add a custom chain to your project. We'll use Base as an example, but you can apply this process to any other chain you want to add.
+Scaffold-ETH 2 uses [viem/chains](https://viem.sh/docs/chains/introduction) as a list of chains.
+Normally, Base already exists in viem/chains and [you can import it and use it](/deploying/deploy-nextjs-app#--targetnetworks), but we're going to add it manually to show you how to do it.
+
+:::info
+Scaffold-ETH 2 consists of two parts:
+
+- `packages/nextjs`: nextjs frontend
+- `packages/hardhat` or `packages/foundry`: hardhat or foundry to deploy smart contracts
+
+The frontend and the hardhat/foundry project use a different set of chains.
+You should add the chain to both the frontend and your hardhat/foundry config.
+
+By doing this, you will be able to deploy the contracts to the chain you added and interact with them from the frontend.
+
+:::
+
+### Step 1: Define the chain
+
+First, create a new file called `customChains.ts` in your `packages/nextjs/utils/` directory.
+
+Open the file with your favorite editor and add the following code to define the chain.
+
+```typescript title="packages/nextjs/utils/customChains.ts"
+import { defineChain } from "viem";
+
+// Base chain
+export const base = defineChain({
+  id: 8453,
+  name: "Base",
+  nativeCurrency: { name: "Base", symbol: "ETH", decimals: 18 },
+  rpcUrls: {
+    default: {
+      http: ["https://mainnet.base.org"],
+    },
+    public: {
+      http: ["https://mainnet.base.org"],
+    },
+  },
+  blockExplorers: {
+    default: {
+      name: "Basescan",
+      url: "https://basescan.org",
+    },
+    etherscan: {
+      name: "Basescan",
+      url: "https://basescan.org",
+    },
+  },
+  contracts: {
+    multicall3: {
+      address: "0xca11bde05977b3631167028862be2a173976ca11",
+      blockCreated: 5022,
+    },
+  },
+});
```

Comment:
Maybe keep this object minimal with only required fields so that people don't get afraid that they have to add fields like `contracts: multicall3` etc. 

<details>

<summary>
Example : 
</summary>

```typescript

// Base chain
export const base = defineChain({
  id: 8453,
  name: "Base",
  nativeCurrency: { name: "Base", symbol: "ETH", decimals: 18 },
  rpcUrls: {
    default: {
      http: ["https://mainnet.base.org"],
    },
  },
  blockExplorers: {
    default: {
      name: "Basescan",
      url: "https://basescan.org",
    },
  },
});

```

</details>

Reply:
Thanks! fixed, i should have thought of that